experimental/Schemes/SpaceGerms.jl

```diff
@@ -48,13 +58,141 @@ A space germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X``
   end
 end
 
+@doc raw"""
+    HypersurfaceGerm{BaseRingType, RingType, SpecType}
+A hypersurface germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X`` of type SpecType and local ring ``O_{(X,x)}`` of type `RingType` over some base ring ``k`` of type `BaseRingType`.
+"""
+@attributes mutable struct HypersurfaceGerm{
+                 BaseRingType<:Ring,
+                 RingType<:Ring,
+                 SpecType<:Spec} <: AbsSpaceGerm{BaseRingType, RingType}
+  X::SpecType
+  f::RingElem
+
+  function HypersurfaceGerm(X::GermAtClosedPoint,f::MPolyLocRingElem; check::Bool=true)
+    base_ring(modulus(OO(X))) == parent(f) || error("baserings do not match")
+    if check
+      (ideal(parent(f),[f]) == modulus(OO(X))) || error("given f does not define given X")
+    end
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X,f)
+  end
+
+## the following is currently unused....
+## as no backend for groebner computations is currently available in this case
+  function HypersurfaceGerm(X::GermAtGeometricPoint, f::MPolyLocRingElem; check::Bool=true)
+    base_ring(modulus(OO(X))) == parent(f) || error("baserings do not match")
+    if check
+      (ideal(parent(f),[f]) == modulus(OO(X))) || error("given f does not define given X")
+    end
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X,f)
+  end
+end
+
+@doc raw"""
+    CompleteIntersectionGerm{BaseRingType, RingType, SpecType}
+A complete intersection germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X`` of type SpecType and local ring ``O_{(X,x)}`` of type `RingType` over some base ring ``k`` of type `BaseRingType`.
+"""
+@attributes mutable struct CompleteIntersectionGerm{BaseRingType<:Ring, RingType<:Ring, SpecType<:Spec} <: AbsSpaceGerm{BaseRingType, RingType}
+  X::SpecType
+  v::Vector{RingElem}
+
+  function CompleteIntersectionGerm(X::GermAtClosedPoint, v::Vector{T}; check::Bool=true) where T<:MPolyLocRingElem
+    R = base_ring(modulus(OO(X)))
+    all(x->parent(x) == R, v) || error("base_rings do not coincide")
+    if check
+## TODO: change dim(Spec(R)) to dim(R) as soon as the fixes for dim have
+## been moved to the algebra side!!!!
+      length(v) == dim(Spec(R)) - dim(X) || error("not a complete intersection")
+      modulus(OO(X)) == ideal(R,v) || error("given tuple does not generate modulus")
+    end
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X,v)
+  end
+
+## the following one is currently unused...
+## as no backend for groebner computations is currently available in this case
```

Comment:
Same as above: We do have localizations at complements of prime ideals since a long time.